### ISSUE TITLE
Decoding produces incorrect results for base64 strings containing null bytes.

### ISSUE THREAD

**Issue by rjcrjc** [gofer/libbase64#5]
Minimal example:
```c++
#include <iostream>
#include <base64.hpp>

int main(int argc, const char **argv)
{
  std::string str("\0\0Xabcdef", 9);
  std::string enc = Base64::encode(str);
  std::cout << enc << std::endl;
  std::string dec = Base64::decode(enc);
  std::cout << dec << std::endl;

  return 0;
}
```
The encode correctly turns the null bytes into 'A' but the decode loses those bytes (as well as the 'X' character in the example). 
What seems to be happening is that the decode makes a string out of every three bytes and C++ strings truncate it early because it contains a null byte.

**Comment by gofer** [comment]
This bug come from my misunderstand in C++ standard library (`std::string`).
If the string contains null characters, `std::string`'s methods instantly return when it faces a null character.
We immediately fix this issue. Thank you for your report.

**Comment by gofer** [comment]
This bug has fixed by pr #6.
